packages/gatsby/src/utils/validate-page-component.ts

```diff
@@ -79,11 +79,13 @@ export function validatePageComponent(
       fileContent.includes(`module.exports`) ||
       fileContent.includes(`exports.default`) ||
       fileContent.includes(`exports["default"]`) ||
-      fileContent.match(/export \{.* as default.*\}/s) ||
-      // this check only applies to js and ts, not mdx
-      /\.(jsx?|tsx?)/.test(path.extname(component))
+      fileContent.match(/export \{.* as default.*\}/s)
 
-    if (!includesDefaultExport) {
+    // this check only applies to js and ts, not mdx
+    if (
+      /\.(jsx?|tsx?)/.test(path.extname(component)) &&
+      !includesDefaultExport
```

Comment:
did it speedup;
- when at first the `(jsx?|tsx?)/.test` is done 
- and then only if it match the calculation for `includesDefaultExport` with the many `manytext.includes` are calculated ?

Reply:
@muescha Yes, that's correct. Performing `includesDefaultExport` before extension check is unnecessary.  Fixed it with PR #24837